pkgs/development/mobile/androidenv/compose-android-packages.nix

```diff
@@ -71,9 +97,21 @@ let
     lib.attrsets.mapAttrsRecursive
       (path: value:
         if (builtins.elemAt path ((builtins.length path) - 1)) == "archives" then
-          (builtins.listToAttrs
-            (builtins.map
-              (archive: lib.attrsets.nameValuePair archive.os (fetchurl { inherit (archive) url sha1; })) value))
+          let
+            validArchives = builtins.filter (archive:
+              let
+                isTargetOs = if builtins.hasAttr "os" archive then
+                  archive.os == os || archive.os == "all" else true;
+                isTargetArc = if builtins.hasAttr "arch" archive then
+                  archive.arch == arch || archive.arch == "all" else true;
+              in
+                isTargetOs && isTargetArc
+            ) value;
+          in
+          assert lib.assertMsg (builtins.length validArchives > 0) "No valid archives for ${lib.concatMapStringsSep "." (x: ''"${x}"'') path} for os=${os}, arch=${arch}";
+          lib.last (map (archive:
```

Comment:
Would it be easier to replace this with null if there's no supported archive, instead of doing this assert check? That way we don't need any external logic. If there's no supported binary for the platform we can use that as the predicate.

Reply:
Sorry, but in my mind it's clear to have an assertion over a null. However, feel free to send the patch and I'll apply it.

Reply:
Fair. I wonder if we'd have to tryEval or something.

In any case, I think if we'd like to submit it this way after @rosscomputerguy checks on aarch64 that's fine with me as it'll unblock people.